src/Model/Resolver/SaveCartItem.php

```diff
@@ -183,6 +187,25 @@ private function setBundleRequestOptions(Product $product, array $data): array
         $data['bundle_option'] = $options;
         return $data;
     }
+
+    /**
+     * @param Product $product
+     * @param array $options
+     * @param array $data
+     */
+    private function setCustomizableOptions(Product $product, array $options, array &$data): void
+    {
+        /**
+         * @TODO Validate if custom options data are correct using @var $product
+         */
```

Comment:
It seems very strange to me, that M2 does not validate these options automatically. Can you please investigate, maybe we do not dispatch some event? I am not sure here!

Except this "TODO" part, everything else seems legit!

Reply:
I also thought the same. Maybe since the scope is `graphql` and not `frontend` so some events might not be firing. But, I have tested the same in Magento Luma as well, and it still does not validate. We could add the validation logic manually though. What are your thoughts on this?

Reply:
I think we should leave the TODO, not implementing validation is OK if M2 does not do it.